<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>mainPageClass</class>
 <widget class="QMainWindow" name="mainPageClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>811</width>
    <height>711</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Root Viwer</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="maximumSize">
    <size>
     <width>16777215</width>
     <height>16777215</height>
    </size>
   </property>
   <layout class="QGridLayout" name="gridLayout_81">
    <item row="0" column="1">
     <widget class="glArea" name="openGLWidget"/>
    </item>
    <item row="1" column="1">
     <layout class="QHBoxLayout" name="glLayout" stretch="0,0">
      <property name="sizeConstraint">
       <enum>QLayout::SetDefaultConstraint</enum>
      </property>
      <item>
       <widget class="QTabWidget" name="tabWidget">
        <property name="maximumSize">
         <size>
          <width>200</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="font">
         <font>
          <family>Segoe UI</family>
          <pointsize>12</pointsize>
          <weight>50</weight>
          <bold>false</bold>
         </font>
        </property>
        <property name="autoFillBackground">
         <bool>false</bool>
        </property>
        <property name="styleSheet">
         <string notr="true">QTabWidget{
	background-color: #C0C0C0;
	border: 1 px solid gray;
}</string>
        </property>
        <property name="currentIndex">
         <number>0</number>
        </property>
        <widget class="QWidget" name="skeletonTab">
         <property name="styleSheet">
          <string notr="true">QTabWidget{
	background-color: #C0C0C0;
	border: 1 px solid gray;
}</string>
         </property>
         <attribute name="title">
          <string>Skeleton</string>
         </attribute>
         <widget class="QCheckBox" name="showSkel">
          <property name="geometry">
           <rect>
            <x>20</x>
            <y>10</y>
            <width>131</width>
            <height>21</height>
           </rect>
          </property>
          <property name="text">
           <string>Show Skeleton</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
         <widget class="QSlider" name="skelWidthSlider">
          <property name="geometry">
           <rect>
            <x>20</x>
            <y>90</y>
            <width>141</width>
            <height>22</height>
           </rect>
          </property>
          <property name="value">
           <number>81</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
         <widget class="QLabel" name="label_2">
          <property name="geometry">
           <rect>
            <x>30</x>
            <y>110</y>
            <width>111</width>
            <height>31</height>
           </rect>
          </property>
          <property name="text">
           <string>Skeleton Width</string>
          </property>
         </widget>
         <widget class="QDoubleSpinBox" name="jetMinBox">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>190</y>
            <width>62</width>
            <height>22</height>
           </rect>
          </property>
         </widget>
         <widget class="QDoubleSpinBox" name="jetMaxBox">
          <property name="geometry">
           <rect>
            <x>90</x>
            <y>190</y>
            <width>62</width>
            <height>22</height>
           </rect>
          </property>
          <property name="value">
           <double>5.000000000000000</double>
          </property>
         </widget>
         <widget class="QPushButton" name="skelFileBrowseBtn">
          <property name="geometry">
           <rect>
            <x>90</x>
            <y>40</y>
            <width>75</width>
            <height>23</height>
           </rect>
          </property>
          <property name="text">
           <string>Select .ply</string>
          </property>
         </widget>
         <widget class="QLineEdit" name="skelFileLabel">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>40</y>
            <width>71</width>
            <height>20</height>
           </rect>
          </property>
         </widget>
         <widget class="QCheckBox" name="whorlAbove">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>440</y>
            <width>151</width>
            <height>31</height>
           </rect>
          </property>
          <property name="text">
           <string>Show Whorls Above</string>
          </property>
          <property name="checked">
           <bool>false</bool>
          </property>
         </widget>
         <widget class="QCheckBox" name="showPlane">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>410</y>
            <width>151</width>
            <height>31</height>
           </rect>
          </property>
          <property name="text">
           <string>Show Soil Plane</string>
          </property>
          <property name="checked">
           <bool>false</bool>
          </property>
         </widget>
         <widget class="QCheckBox" name="showLvl0Root">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>260</y>
            <width>151</width>
            <height>31</height>
           </rect>
          </property>
          <property name="text">
           <string>Show Stem Path</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
         <widget class="QCheckBox" name="showLvl1Root">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>290</y>
            <width>151</width>
            <height>31</height>
           </rect>
          </property>
          <property name="text">
           <string>Show Stem Region</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
         <widget class="QCheckBox" name="showLvl2Root">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>320</y>
            <width>151</width>
            <height>31</height>
           </rect>
          </property>
          <property name="text">
           <string>Show Nodal Roots</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
         <widget class="QCheckBox" name="showLvl3Root">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>350</y>
            <width>151</width>
            <height>31</height>
           </rect>
          </property>
          <property name="text">
           <string>Show Secondary Roots</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
         <widget class="QCheckBox" name="showLvlLaterRoot">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>380</y>
            <width>151</width>
            <height>31</height>
           </rect>
          </property>
          <property name="text">
           <string>Show Other Roots</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
         <widget class="QCheckBox" name="whorlBelow">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>470</y>
            <width>151</width>
            <height>31</height>
           </rect>
          </property>
          <property name="text">
           <string>Show Whorls Below</string>
          </property>
          <property name="checked">
           <bool>false</bool>
          </property>
         </widget>
         <widget class="QCheckBox" name="nodalRootAbove">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>500</y>
            <width>151</width>
            <height>31</height>
           </rect>
          </property>
          <property name="text">
           <string>Show Nodal Roots Above</string>
          </property>
          <property name="checked">
           <bool>false</bool>
          </property>
         </widget>
         <widget class="QCheckBox" name="nodalRootBelow">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>530</y>
            <width>151</width>
            <height>31</height>
           </rect>
          </property>
          <property name="text">
           <string>Show Nodal Roots Below</string>
          </property>
          <property name="checked">
           <bool>false</bool>
          </property>
         </widget>
         <widget class="QLineEdit" name="annotationFileLabel">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>60</y>
            <width>71</width>
            <height>20</height>
           </rect>
          </property>
         </widget>
         <widget class="QPushButton" name="annotationFileBrowseBtn">
          <property name="geometry">
           <rect>
            <x>90</x>
            <y>60</y>
            <width>75</width>
            <height>23</height>
           </rect>
          </property>
          <property name="text">
           <string>Select .txt</string>
          </property>
         </widget>
         <widget class="QComboBox" name="skeletonColor">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>140</y>
            <width>141</width>
            <height>21</height>
           </rect>
          </property>
          <property name="editable">
           <bool>false</bool>
          </property>
          <item>
           <property name="text">
            <string>Normal Coloring(Black)</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Color by Hierarchy</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Color by Radius</string>
           </property>
          </item>
         </widget>
         <widget class="QLabel" name="label_7">
          <property name="geometry">
           <rect>
            <x>30</x>
            <y>160</y>
            <width>111</width>
            <height>31</height>
           </rect>
          </property>
          <property name="text">
           <string>Skeleton Color</string>
          </property>
         </widget>
         <widget class="QLabel" name="label_8">
          <property name="geometry">
           <rect>
            <x>20</x>
            <y>220</y>
            <width>111</width>
            <height>31</height>
           </rect>
          </property>
          <property name="text">
           <string>Skeleton Color Range</string>
          </property>
         </widget>
         <widget class="QCheckBox" name="editOn">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>560</y>
            <width>74</width>
            <height>20</height>
           </rect>
          </property>
          <property name="text">
           <string>EditOn</string>
          </property>
         </widget>
         <widget class="QTextEdit" name="inputText">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>580</y>
            <width>131</width>
            <height>31</height>
           </rect>
          </property>
         </widget>
         <widget class="QPushButton" name="inputValue">
          <property name="geometry">
           <rect>
            <x>150</x>
            <y>580</y>
            <width>31</width>
            <height>28</height>
           </rect>
          </property>
          <property name="text">
           <string>Go</string>
          </property>
         </widget>
         <widget class="QPushButton" name="swapLast">
          <property name="geometry">
           <rect>
            <x>110</x>
            <y>560</y>
            <width>61</width>
            <height>21</height>
           </rect>
          </property>
          <property name="text">
           <string>Swap</string>
          </property>
         </widget>
        </widget>
        <widget class="QWidget" name="meshTab">
         <property name="styleSheet">
          <string notr="true">QTabWidget{
	background-color: #C0C0C0;
	border: 1 px solid gray;
}</string>
         </property>
         <attribute name="title">
          <string>Mesh    </string>
         </attribute>
         <widget class="QCheckBox" name="showMesh">
          <property name="geometry">
           <rect>
            <x>30</x>
            <y>10</y>
            <width>131</width>
            <height>21</height>
           </rect>
          </property>
          <property name="text">
           <string>Show Mesh</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
         <widget class="QSlider" name="alphaSlider">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>90</y>
            <width>141</width>
            <height>22</height>
           </rect>
          </property>
          <property name="value">
           <number>55</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
         <widget class="QLabel" name="label">
          <property name="geometry">
           <rect>
            <x>60</x>
            <y>110</y>
            <width>111</width>
            <height>31</height>
           </rect>
          </property>
          <property name="text">
           <string>Transparency</string>
          </property>
         </widget>
         <widget class="QSlider" name="meshColorRedSlider">
          <property name="geometry">
           <rect>
            <x>20</x>
            <y>140</y>
            <width>141</width>
            <height>22</height>
           </rect>
          </property>
          <property name="value">
           <number>99</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
         <widget class="QSlider" name="meshColorGreenSlider">
          <property name="geometry">
           <rect>
            <x>20</x>
            <y>160</y>
            <width>141</width>
            <height>22</height>
           </rect>
          </property>
          <property name="value">
           <number>99</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
         <widget class="QSlider" name="meshColorBlueSlider">
          <property name="geometry">
           <rect>
            <x>20</x>
            <y>180</y>
            <width>141</width>
            <height>22</height>
           </rect>
          </property>
          <property name="value">
           <number>99</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
         <widget class="QLabel" name="label_3">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>130</y>
            <width>16</width>
            <height>31</height>
           </rect>
          </property>
          <property name="text">
           <string>R</string>
          </property>
         </widget>
         <widget class="QLabel" name="label_4">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>150</y>
            <width>16</width>
            <height>31</height>
           </rect>
          </property>
          <property name="text">
           <string>G</string>
          </property>
         </widget>
         <widget class="QLabel" name="label_5">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>170</y>
            <width>16</width>
            <height>31</height>
           </rect>
          </property>
          <property name="text">
           <string>B</string>
          </property>
         </widget>
         <widget class="QLabel" name="label_6">
          <property name="geometry">
           <rect>
            <x>70</x>
            <y>190</y>
            <width>41</width>
            <height>31</height>
           </rect>
          </property>
          <property name="text">
           <string>Color</string>
          </property>
         </widget>
         <widget class="QLineEdit" name="meshFileLabel">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>30</y>
            <width>71</width>
            <height>20</height>
           </rect>
          </property>
         </widget>
         <widget class="QPushButton" name="meshFileBrowseBtn">
          <property name="geometry">
           <rect>
            <x>90</x>
            <y>30</y>
            <width>75</width>
            <height>23</height>
           </rect>
          </property>
          <property name="text">
           <string>Select .off</string>
          </property>
         </widget>
         <widget class="QLineEdit" name="meshFileLabel_obj">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>60</y>
            <width>71</width>
            <height>20</height>
           </rect>
          </property>
         </widget>
         <widget class="QPushButton" name="meshFileBrowseBtn_obj">
          <property name="geometry">
           <rect>
            <x>90</x>
            <y>60</y>
            <width>75</width>
            <height>23</height>
           </rect>
          </property>
          <property name="text">
           <string>Select .obj</string>
          </property>
         </widget>
         <widget class="QSlider" name="backColorRedSlider">
          <property name="geometry">
           <rect>
            <x>20</x>
            <y>230</y>
            <width>141</width>
            <height>22</height>
           </rect>
          </property>
          <property name="value">
           <number>99</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
         <widget class="QSlider" name="backColorBlueSlider">
          <property name="geometry">
           <rect>
            <x>20</x>
            <y>270</y>
            <width>141</width>
            <height>22</height>
           </rect>
          </property>
          <property name="value">
           <number>99</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
         <widget class="QSlider" name="backColorGreenSlider">
          <property name="geometry">
           <rect>
            <x>20</x>
            <y>250</y>
            <width>141</width>
            <height>22</height>
           </rect>
          </property>
          <property name="value">
           <number>99</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
         <widget class="QLabel" name="label_9">
          <property name="geometry">
           <rect>
            <x>50</x>
            <y>290</y>
            <width>101</width>
            <height>31</height>
           </rect>
          </property>
          <property name="text">
           <string>Background Color</string>
          </property>
         </widget>
         <widget class="QLabel" name="label_10">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>240</y>
            <width>16</width>
            <height>31</height>
           </rect>
          </property>
          <property name="text">
           <string>G</string>
          </property>
         </widget>
         <widget class="QLabel" name="label_11">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>220</y>
            <width>16</width>
            <height>31</height>
           </rect>
          </property>
          <property name="text">
           <string>R</string>
          </property>
         </widget>
         <widget class="QLabel" name="label_12">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>260</y>
            <width>16</width>
            <height>31</height>
           </rect>
          </property>
          <property name="text">
           <string>B</string>
          </property>
         </widget>
        </widget>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_2">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeType">
         <enum>QSizePolicy::Expanding</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>glArea</class>
   <extends>QOpenGLWidget</extends>
   <header>glarea.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="mainPage.qrc"/>
 </resources>
 <connections/>
</ui>
